### ISSUE TITLE
Colab out of CPU memory crash "^C" using 7b FP16-sharded LLM

### ISSUE THREAD

**Issue by quantumalchemy** [huggingface/autotrain-advanced#277]
Getting out of CPU ram using colab  12.7 GB (free)
^C

## CPU RAM SPIKE after model Was loaded and 
!autotrain llm \
--train \ executed
** Runtime using: T4 GPU selected
## OUT OF CPU MEMORY - NO other log or ERROR just stops

Was working last week Ok .. * seems loading in cpu And Not loading in GPU 
re-create --> 
https://colab.research.google.com/drive/13KrQUCMR47AOzOnAQbI9HxHkBFZyrwy4?usp=sharing

>>
#@markdown #### Project Config
#@markdown Note: if you are using a restricted/private model, you need to enter your Hugging Face token in the next step.
project_name = 'autotrain_llm' # @param {type:"string"}
model_name = 'AyyYOO/Luna-AI-Llama2-Uncensored-FP16-sharded' # @param {type:"string"}

#@markdown ---
#@markdown #### Push to Hub?
#@markdown Use these only if you want to push your trained model to a private repo in your Hugging Face Account
#@markdown If you dont use these, the model will be saved in Google Colab and you are required to download it manually.
#@markdown Please enter your Hugging Face write token. The trained model will be saved to your Hugging Face account.
#@markdown You can find your token here: https://huggingface.co/settings/tokens
push_to_hub = False # @param ["False", "True"] {type:"raw"}
hf_token = "hf_XXX" #@param {type:"string"}
repo_id = "username/repo_name" #@param {type:"string"}

#@markdown ---
#@markdown #### Hyperparameters
learning_rate = 2e-4 # @param {type:"number"}
num_epochs = 6 #@param {type:"number"}
batch_size = 2 # @param {type:"slider", min:1, max:32, step:1}
block_size = 1046 # @param {type:"number"}
trainer = "sft" # @param ["default", "sft"] {type:"raw"}
warmup_ratio = 0.1 # @param {type:"number"}
weight_decay = 0.01 # @param {type:"number"}
gradient_accumulation = 4 # @param {type:"number"}
use_fp16 = True # @param ["False", "True"] {type:"raw"}
use_peft = True # @param ["False", "True"] {type:"raw"}
use_int4 = True # @param ["False", "True"] {type:"raw"}
lora_r = 32 #@param {type:"number"}
lora_alpha = 64 #@param {type:"number"}
lora_dropout = 0.05 #@param {type:"number"}

!autotrain llm \
--train \
--model ${MODEL_NAME} \
--project-name ${PROJECT_NAME} \
--data-path /content/LLM/MyDrive/LLM/data/ \
--text-column text \
--lr ${LEARNING_RATE} \
--batch-size ${BATCH_SIZE} \
--epochs ${NUM_EPOCHS} \
--block-size ${BLOCK_SIZE} \
--warmup-ratio ${WARMUP_RATIO} \
--lora-r ${LORA_R} \
--lora-alpha ${LORA_ALPHA} \
--lora-dropout ${LORA_DROPOUT} \
--weight-decay ${WEIGHT_DECAY} \
#--merge-adapter \
--gradient-accumulation ${GRADIENT_ACCUMULATION} \
$( [[ "$USE_FP16" == "True" ]] && echo "--fp16" ) \
$( [[ "$USE_PEFT" == "True" ]] && echo "--use-peft" ) \
$( [[ "$USE_INT4" == "True" ]] && echo "--use-int4" ) \
$( [[ "$PUSH_TO_HUB" == "True" ]] && echo "--push-to-hub --token ${HF_TOKEN} --repo-id ${REPO_ID}" )

OUTPUT>> ...
Downloading (…)l-00027-of-00027.bin: 100% 352M/352M [02:29<00:00, 2.36MB/s]
Downloading shards: 100% 27/27 [40:59<00:00, 91.09s/it] 
^C

## OUT OF CPU MEMORY - NO other log or ERROR just stops

**Comment by vk-maurya** [comment]
I was also facing the same problem autotrain command was not properly called. You can use the below format

```
!autotrain llm \
--train \
--model ${MODEL_NAME} \
--project-name ${PROJECT_NAME} \
--data-path /content/data/ \
--text-column text \
--use_int4 \
--lr ${LEARNING_RATE} \
--batch-size ${BATCH_SIZE} \
--epochs ${NUM_EPOCHS} \
--block-size ${BLOCK_SIZE} \
--warmup-ratio ${WARMUP_RATIO} \
--lora-r ${LORA_R} \
--lora-alpha ${LORA_ALPHA} \
--lora-dropout ${LORA_DROPOUT} \
--weight-decay ${WEIGHT_DECAY} \
--gradient-accumulation ${GRADIENT_ACCUMULATION} \
$( [[ "$USE_FP16" == "True" ]] && echo "--fp16" ) \
$( [[ "$USE_PEFT" == "True" ]] && echo "--use-peft" ) \
$( [[ "$USE_INT4" == "True" ]] && echo "--use-int4" ) 
```
